src/tbp/monty/frameworks/models/evidence_matching/learning_module.py

```diff
@@ -718,6 +724,7 @@ def _update_possible_matches(self, query):
         # NOTE: would not need to do this if we are still voting
         # Call this update in the step method?
         self.possible_matches = self._threshold_possible_matches()
+        self.previous_mlh = copy.deepcopy(self.current_mlh)
         self.current_mlh = self._calculate_most_likely_hypothesis()
```

Comment:
Do we need the `deepcopy` here? On line 728, we're assigning a new value to `current_mlh`, so after these two lines (without the `deepcopy`) we'd have:

self.previous_mlh pointing to the old value of current_mlh
self.current_mlh pointing to the new value created inside _calculate_most_likely_hypothesis

I think the `deepcopy` call is just extra unnecessary work here.
```suggestion
        self.previous_mlh = self.current_mlh
        self.current_mlh = self._calculate_most_likely_hypothesis()
```